app/src/main/java/org/coepi/android/ui/home/HomeFragment.kt

```diff
@@ -26,7 +28,20 @@ class HomeFragment : Fragment() {
             val intent = Intent(Intent.ACTION_VIEW, webpage)
             startActivity(intent)
         }
-    }.root
 
+        homeCardsRecyclerView.apply {
+            setHasFixedSize(true)
+            layoutManager = LinearLayoutManager(context, VERTICAL, false)
+        }
+
+        val adapter = HomeAdapter(onItemClicked = { item ->
+            viewModel.onClicked(item)
+        })
+        homeCardsRecyclerView.adapter = adapter
+        //adapter.submitList(viewModel.homeCardItems)
 
+        viewModel.homeCardObservable.observeWith(viewLifecycleOwner) {
```

Comment:
@i-schuetz, so @mellowgeek worked with me for over an hour and we got pretty far making this home card screen reactive, but for some reason it seems like the "startsWith" on our observable isnt working or something because the cards arent being populated on start. We did add an observable to the view model though. I'm not sure what's not working here. Do you have any suggestions?

Reply:
You have to use the `startsWith` with `()`. The one with the closure does not emit the returned item.

Also:
- It's better to put the `startsWith` on the alerts: "No alerts yet" is the state we want to start with:
```kotlin
.startWith(emptyList<List<Alert>>())
```
- We should avoid using mutability. You could write `map` like this:
```kotlin
map { alerts ->
    homeCardItems.map {
        if (it.cardId == SEE_ALERTS) {
            it.copy(newAlerts = alerts.isNotEmpty(), newAlertsTitle = title(alerts.size))
        } else {
            it
        }
    }
}
```
This just generates a new list. Then you can leave `homeCardItems` immutable. It should be private too.

Reply:
Oh and we need `distinctUntilChanged` on the alerts too.

Reply:
thanks @i-schuetz that worked! will update this PR soon